sdk/storage/Azure.Storage.Files.Shares/src/ShareDirectoryClient.cs

```diff
@@ -1840,12 +1899,70 @@ await GetSubdirectoryClient(subdirectoryName)
                 cancellationToken).ConfigureAwait(false);
             return Response.FromValue(file, response.GetRawResponse());
         }
+
+        /// <summary>
+        /// Creates a new file or replaces an existing file.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-file"/>.
+        /// </summary>
+        /// <remarks>
+        /// This method only initializes the file.
+        /// To add content, use <see cref="ShareFileClient.UploadRange(ShareFileRangeWriteType, HttpRange, System.IO.Stream, byte[], IProgress{long}, FileRequestConditions, CancellationToken)"/>.
+        /// </remarks>
+        /// <param name="fileName">The name of the file.</param>
+        /// <param name="maxSize">
+        /// Required. Specifies the maximum size for the file.
+        /// </param>
+        /// <param name="httpHeaders">
+        /// Optional standard HTTP header properties that can be set for the file.
+        /// </param>
+        /// <param name="metadata">
+        /// Optional custom metadata to set for the file.
+        /// </param>
+        /// <param name="smbProperties">
+        /// Optional SMB properties to set for the file.
+        /// </param>
+        /// <param name="filePermission">
+        /// Optional file permission to set for the file.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{FileClient}"/> referencing the file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        [ForwardsClientCalls]
+        public virtual async Task<Response<ShareFileClient>> CreateFileAsync(
+            string fileName,
+            long maxSize,
+            ShareFileHttpHeaders httpHeaders,
+            Metadata metadata,
+            FileSmbProperties smbProperties,
+            string filePermission,
+            CancellationToken cancellationToken = default)
```

Comment:
Why is this optional on async version but required on sync version?

Reply:
fixed.